css/selectors/focus-visible-013.html

```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html>
+<meta charset="utf-8" />
+<title>CSS Test (Selectors): Mouse focus does not match :focus-visible</title>
+<link rel="author" title="Manuel Rego Casasnovas" href="mailto:rego@igalia.com">
+<link rel="help" href="https://drafts.csswg.org/selectors-4/#the-focus-visible-pseudo" />
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<style>
+  @supports not (selector(:focus-visible)) {
+    :focus {
+      background-color: red;
+    }
+  }
+
+  :focus-visible {
+    background-color: red;
+  }
+
+  :focus:not(:focus-visible) {
+    background-color: lime;
+  }
+</style>
+
+<p>This test checks that using the mouse to focus an element does not trigger <code>:focus-visible</code> matching, and the element doesn't show any focus ring.</p>
```

Comment:
The first part of this is already well covered by existing tests; what exactly is this test checking?

Reply:
This test current fails in Chromium and passes in Firefox. So it's not totally covered by other tests.

This checks that when you focus via mouse click, the element doesn't get a focus ring. As it doesn't match `:focus-visble` it shouldn't get any outline by default. The default UA style sheet rule is going to be changed from `:focus { outline: auto; }` to `:focus-visible { outline: auto; }`, so Chromium shouldn't show any outline by default when `:focus-visible` doesn't match, like in this test.

Reply:
Got it, thanks for the explanation.

That being the case, I think we don't need the custom styles (red/green background), since the selector matching part is already covered by other tests.

Instead, we could reword the test as something like "... by default, using the mouse to focus a generic element does not show a focus ring". Then, we could just check the outline style on focus. If the selector isn't supported, or isn't the UA default, this will fail.

Reply:
I agree that the customs styles are not needed for the test, but I think they're useful as it's covered in other tests. But I don't think it's bad to have them, as that makes easier to identify a failure if `:focus-visible` is not supported at all.

I've reworded things as suggested but I'll keep the background colors unless you see a big problem with them.

Reply:
No problem if you strongly prefer to keep the background colours, but why not just have something like the following?

```css
.warning {
  display: none;
}

:supports not (:focus-visible) {
  .warning {
    display: initial;
  }
}
```

```html
<div class="warning">
<code>:focus-visible</code> is not supported. This test will not pass.
</div>
```

Reply:
The problem with that warning is that it won't change anything on the test, and maybe the script tests could be true without any `:focus-visible` support at all. Probably not this particular test but it might happen in other tests.

I want that a browser that doesn't support `:focus-visible` at all, fail on the automatic tests. So using this kind of red outlines/backgrounds is usually useful for both the automatic testing and manual testing.

Also I've added them in most of the test on the suite in previous patches (before that some tests were passing in WebKit without any kind of implementation), so maybe if we want to change that we could do for all of them together in follow-up patches.

Reply:
Ok, maybe just add a comment explaining what the background colours are testing then, so we don't accidentally remove them later.